Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/ThresholdedRandomCutForest.java

```diff
@@ -290,24 +561,27 @@ protected boolean trigger(DiVector candidate, int gap, int baseDimension, DiVect
      * @return description containing scores, grade, confidence, expected values,
      *         attribution etc.
      */
-    protected AnomalyDescriptor getAnomalyDescription(double[] point) {
+    protected AnomalyDescriptor getAnomalyDescription(double[] point, long inputTimeStamp, double[] inputPoint) {
```

Comment:
This is a very long method with a lot of logic doing a lot of different things. I wonder if we can break this down, which would also make it easier to unit test.